### PR TITLE
Decouple excheck from datapackage and rest

### PR DESCRIPTION
**Pull request by dlc-ariel**

### CODE REVIEW

File: FreeTAKServer/controllers/ExCheckController.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+from datetime import datetime
+from os import listdir
+from pathlib import Path, PurePath
+from xml.etree.ElementTree import Element
+
+from defusedxml import ElementTree as etree
+from flask import request
+from FreeTAKServer.controllers.configuration.LoggingConstants import LoggingConstants
+from FreeTAKServer.controllers.configuration.MainConfig import MainConfig
+from FreeTAKServer.controllers.CreateLoggerController import CreateLoggerController
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:14]: > __[Pylint]__ No name 'CreateLoggerController' in module 'FreeTAKServer.controllers' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357389))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `no-name-in-module` | `error` | E0611 | FreeTAKServer.controllers.ExCheckController | - |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357389) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

check out this class

Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py]: > __[Pylint]__ No name 'CreateLoggerController' in module 'FreeTAKServer.controllers' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357389))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `no-name-in-module` | `error` | E0611 | FreeTAKServer.controllers.ExCheckController | - |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357389) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

missing CreateLoggerController

File: FreeTAKServer/controllers/ExCheckController.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+from FreeTAKServer.controllers.DatabaseControllers.DatabaseController import DatabaseController
+from FreeTAKServer.controllers.ExCheckControllers.templateToJsonSerializer import templateSerializer
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py]: > __[Pylint]__ No name 'ExCheckControllers' in module 'FreeTAKServer.controllers' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357391))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `no-name-in-module` | `error` | E0611 | FreeTAKServer.controllers.ExCheckController | - |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357391) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

fix import

File: FreeTAKServer/controllers/ExCheckController.py [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+            # this is where the client will post the xmi of a template
+            # possibly the uid of the client submitting the template
+            Y = request
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:55]: > __[Pylint]__ Unused variable 'Y' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357380))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `unused-variable` | `warning` | W0612 | FreeTAKServer.controllers.ExCheckController | ExCheckController.template |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357380) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

remove unused

File: FreeTAKServer/controllers/ExCheckController.py [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+    def exchecktemplates(self):
+        try:
+            return templateSerializer().convert_object_to_json(DatabaseController().query_ExCheck())
+        except Exception as e:
+            self.logger.error(str(e))
+
+    def template(self, pipe):
+        try:
+            # TODO remove hard coded string
```
[the remark this answers is not in the document]
Comment (dlc-ariel) [reply FreeTAKServer/controllers/ExCheckController.py:51]: @naman108, is the solution to this using an ET.Element and building it using that class?

File: FreeTAKServer/controllers/ExCheckController.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+            # possibly the uid of the client submitting the template
+            uid = request.args.get('clientUid')
+            XMI = request.data.decode()
+            PIPE = pipe
+            serializer = templateSerializer(XMI)
+            object = serializer.convert_template_to_object()
+            object.timestamp = datetime.strptime(object.timestamp, "%Y-%m-%dT%H:%M:%S.%fZ")
+            serializer.create_DB_object(object)
+            xml = etree.fromstring(XMI)
+            tasks = xml.find('checklistTasks')
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:63]: > __[Pylint]__ Unused variable 'tasks' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357381))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `unused-variable` | `warning` | W0612 | FreeTAKServer.controllers.ExCheckController | ExCheckController.template |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357381) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

remove unused variable

File: FreeTAKServer/controllers/ExCheckController.py [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+            resources.find('filename').text = temp.find('checklistDetails').find('uid').text + '.xml'
+            resources.findall('keywords')[0].text = temp.find('checklistDetails').find('name').text
+            resources.findall('keywords')[1].text = temp.find('checklistDetails').find('description').text
+            resources.findall('keywords')[2].text = temp.find('checklistDetails').find('creatorCallsign').text
+            resources.find('uid').text = temp.find('checklistDetails').find('uid').text
+            resources.find('name').text = temp.find('checklistDetails').find('uid').text
+            resources.find('size').text = str(len(XMI))
+            resources.find('hash').text = str(hashlib.sha256(str(XMI).encode()).hexdigest())
+            z = etree.tostring(cot)
+            from FreeTAKServer.model.testobj import testobj
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:86]: > __[Pylint]__ Import outside toplevel (FreeTAKServer.model.testobj.testobj) ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357274))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `import-outside-toplevel` | `convention` | C0415 | FreeTAKServer.controllers.ExCheckController | ExCheckController.template |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357274) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

move import in class head

File: FreeTAKServer/controllers/ExCheckController.py [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+
+        tasks = xml.find('checklistTasks')
+        for task in tasks:
+            taskuid = task.find('uid')
+            taskuid.text = str(uuid.uuid4())
+
+        with open(
+                str(PurePath(Path(config.ExCheckChecklistFilePath), Path(uid + '.xml'))),
+                'w+') as file:
+            y = etree.tostring(xml)
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:139]: > __[Pylint]__ Unused variable 'y' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357383))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `unused-variable` | `warning` | W0612 | FreeTAKServer.controllers.ExCheckController | ExCheckController.startList |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357383) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

control not used variable
y = etree.tostring(xml)

File: FreeTAKServer/controllers/ExCheckController.py [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+            description=request.args.get('description'),
+            callsign=request.args.get('callsign'),
+            name=request.args.get('name'), uid=uid,
+            filename=f'{uid}.xml', template=excheckobj
+        )
+
+        return str(open(str(PurePath(Path(config.ExCheckChecklistFilePath), Path(uid + '.xml'))), 'r').read()), 200
+
+    def update_checklist(self):
+        r = request
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:156]: > __[Flake8]__ local variable 'r' is assigned to but never used ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357254))
>
> <details>
>
> | Rule |
> | ---- |
> | `F841` |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357254) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

review unused variable

File: FreeTAKServer/controllers/ExCheckController.py [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+        xml = XMLCoTController().serialize_model_to_CoT(object)
+        rawcot.xmlString = xml
+
+        PIPE.put(rawcot)
+        # PIPE.send()
+
+        return '', 200
+
+    def activechecklists(self):
+        checklists = Checklists.Checklist()
```
Comment (brothercorvo) [inline FreeTAKServer/controllers/ExCheckController.py:248]: > __[Pylint]__ Unused variable 'checklists' ([view](https://sider.review/gh/repos/238457011/pulls/449#issue-8357385))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `unused-variable` | `warning` | W0612 | FreeTAKServer.controllers.ExCheckController | ExCheckController.activechecklists |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/449#issue-8357385) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

unused?

**Comment by brothercorvo** [comment]
@dlc-ariel please address issues above

**Pull request merged**